### ISSUE TITLE
SC2319 - confusing warning message when checking $? after [[  ]]

### ISSUE THREAD

**Issue by aslatter** [koalaman/shellcheck#2544]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2319
- My shellcheck version (`shellcheck --version` or "online"): v0.8.0-45-gd1d574c
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit - (not really, but "--version" indicates I am using the current commit)

#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

[[ -n "${DB_HOST:-}" && -n "${DB_TOKEN:-}" && -n "${DISTRO:-}" ]]
should_check_db=$?

echo "$should_check_db"

```

#### Here's what shellcheck currently says:

```
In /work/test.sh line 4:
should_check_db=$?
                ^-- SC2319 (warning): This $? refers to a condition, not a command. Assign to a variable to avoid it being overwritten.

For more information:
  https://www.shellcheck.net/wiki/SC2319 -- This $? refers to a condition, no...
```

#### Here's what I wanted or expected to see:

It's confusing that shellcheck is telling me to assign `$?` to a variable on the line where I assign `$?` to a variable.

It's not clear why I shouldn't be checking the exit-code of "[[]]" - that's the whole point of "[[]]". None of the "problematic code" examples in the wiki look like the code I have, also.

Better descriptions or examples around why this construction is problematic would be helpful.

**Comment by koalaman** [comment]
Agreed, your pattern was listed under "Exceptions". However, the check has now been updated to exclude cases where the `$?` is used unconditionally (like your case or `[ -d test ]; return $?`). Thanks for testing on the git branch!

**Comment by aslatter** [comment]
Thanks for the fix! The change works great.

**Comment by megasam-programs** [comment]
Hi all!!

Conducted a test on the site [https://www.shellcheck.net](https://www.shellcheck.net)
I report that this problem is repeated inside a function:

**Here's a snippet or screenshot that shows the problem:**
#!/bin/bash
print_codout() {
	RETURN=$?
	printf '%s\n' "$RETURN"
}
[ -f foo ] || print_codout

**Here's what shellcheck currently says:**
$ shellcheck myscript

 [Line 3:]
        RETURN=$?
                         ^-- SC2319 (warning): This $? refers to a condition, not a command. Assign to a variable to avoid it being overwritten.

**Comment by megasam-programs** [comment]
Hi all!!

Conducted a test on the site https://www.shellcheck.net
I report that this problem is repeated inside a function:

**Here's a snippet or screenshot that shows the problem:**
`#!/bin/bash
print_codout() {
RETURN=$?
printf '%s\n' "$RETURN"
}
[ -f foo ] || print_codout`

**Here's what shellcheck currently says:**
`$ shellcheck myscript

[Line 3:]
RETURN=$?
^-- SC2319 (warning): This $? refers to a condition, not a command. Assign to a variable to avoid it being overwritten.`

**Comment by megasam-programs** [comment]
Hi all!!

Conducted a test on the site https://www.shellcheck.net
I report that this problem is repeated inside a function:

**Here's a snippet or screenshot that shows the problem:**

```#!/bin/bash
print_codout() {
RETURN=$?
printf '%s\n' "$RETURN"
}
[ -f foo ] || print_codout
```

**Here's what shellcheck currently says:**
```$ shellcheck myscript

[Line 3:]
RETURN=$?
^-- SC2319 (warning): This $? refers to a condition, not a command. Assign to a variable to avoid it being overwritten.
```

**Comment by megasam-programs** [comment]
Hi all!!

Conducted a test on the site https://www.shellcheck.net
I report that this problem is repeated inside a function:

**Here's a snippet or screenshot that shows the problem:**

```
#!/bin/bash
print_codout() {
	RETURN=$?
	printf '%s\n' "$RETURN"
}
[ -f foo ] || print_codout
```

**Here's what shellcheck currently says:**
```
$ shellcheck myscript

Line 3
RETURN=$?
       ^-- SC2319 (warning): This $? refers to a condition, not a command. Assign to a variable to avoid it being overwritten.
```
